source/extensions/common/wasm/context.cc

```diff
@@ -1622,6 +1634,13 @@ void Context::onGrpcReceiveWrapper(uint32_t token, ::Envoy::Buffer::InstancePtr
 
 void Context::onGrpcCloseWrapper(uint32_t token, const Grpc::Status::GrpcStatus& status,
                                  const absl::string_view message) {
+  if (proxy_wasm::current_context_ != nullptr) {
+    // We are in a reentrant call, so defer.
+    auto message_string = std::string(message);
+    wasm()->addAfterVmCallAction([this, token, status, message_string] {
```

Comment:
(same here)

```
wasm()->addAfterVmCallAction([this, token, status, message = std::string(message)] {
```

to make it clearer that we're moving this value to lambda?

Reply:
Done.